crates/cxx-qt-gen/src/generator/rust/constructor.rs

```diff
@@ -0,0 +1,310 @@
+// SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Leon Matthes <leon.matthes@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use crate::{
+    generator::{
+        naming::{namespace::NamespaceName, qobject::QObjectName},
+        rust::qobject::GeneratedRustQObjectBlocks,
+    },
+    parser::constructor::Constructor,
+};
+
+use convert_case::{Case, Casing};
+use proc_macro2::{Span, TokenStream};
+use quote::{format_ident, quote};
+use syn::{parse_quote, Ident, Result, Type};
+
+const CONSTRUCTOR_ARGUMENTS: &str = "CxxQtConstructorArguments";
+const BASE_ARGUMENTS: &str = "CxxQtConstructorBaseArguments";
+const NEW_ARGUMENTS: &str = "CxxQtConstructorNewArguments";
+const INITIALIZE_ARGUMENTS: &str = "CxxQtConstructorInitializeArguments";
+
+fn map_types<F: FnMut((usize, &Type)) -> TokenStream>(
+    args: &Option<Vec<Type>>,
+    f: F,
+) -> Vec<TokenStream> {
+    args.as_ref()
+        .map(|args| args.iter().enumerate().map(f).collect())
+        .unwrap_or_default()
+}
+
+fn extract_arguments_from_tuple(args: &Option<Vec<Type>>, tuple_name: Ident) -> Vec<TokenStream> {
+    map_types(args, |(index, _ty)| {
+        let arg_name = format_ident!("arg{index}");
+        let index = syn::LitInt::new(index.to_string().as_str(), Span::call_site());
+        quote! {
+            #arg_name: #tuple_name.#index
+        }
+    })
+}
+
+fn extract_arguments_from_struct(args: &Option<Vec<Type>>, struct_name: Ident) -> Vec<TokenStream> {
+    map_types(args, |(index, _ty)| {
+        let arg_name = format_ident!("arg{index}");
+        quote! {
+            #struct_name.#arg_name
+        }
+    })
+}
+
+fn argument_members(args: &Option<Vec<Type>>) -> Vec<TokenStream> {
+    map_types(args, |(index, ty)| {
+        let arg_name = format_ident!("arg{index}");
+        quote! {
+            #arg_name: #ty
+        }
+    })
+}
+
+fn generate_default_constructor(
+    qobject_idents: &QObjectName,
+    namespace: &NamespaceName,
+) -> GeneratedRustQObjectBlocks {
+    let rust_struct_ident = &qobject_idents.rust_struct.rust;
+    let create_rs_ident = format_ident!(
+        "create_rs_{object_name}",
+        object_name = rust_struct_ident.to_string().to_case(Case::Snake)
+    );
+    let namespace_internals = &namespace.internal;
+
+    GeneratedRustQObjectBlocks {
+        cxx_mod_contents: vec![parse_quote! {
+        extern "Rust" {
+            #[cxx_name = "createRs"]
+            #[namespace = #namespace_internals]
+            fn #create_rs_ident() -> Box<#rust_struct_ident>;
+        }
+        }],
+        cxx_qt_mod_contents: vec![parse_quote! {
+            /// Generated CXX-Qt method which creates a boxed rust struct of a QObject
+            pub fn #create_rs_ident() -> std::boxed::Box<#rust_struct_ident> {
```

Comment:
should probably use `alloc::boxed::Box` instead

Reply:
Done